qt/scientific_interfaces/Muon/MuonAnalysisResultTableCreator.cpp

```diff
@@ -24,7 +26,15 @@ using Mantid::API::WorkspaceFactory;
 using Mantid::API::WorkspaceGroup;
 
 namespace {
-
+/**
+ * Checks if a string is a numeric value
+ * @param string:: string to test
+ * @returns :: bool if it is a string
+ */
+bool isNumber(const std::string &string) {
```

Comment:
Could we add some unit tests to this to help document, what the regex does and does not reject?

Reply:
yes, I can quickly put something together

Reply:
The Windows compiler is warning on the fact that we've included a `.cpp` in a test header and we're trying to then link to something in an anonymous namespace - which you're not supposed to do as they are supposed to be entirely internal. Anonymous methods shouldn't be tested directly as they should be considered implementation details. If something is big/complicated enough to require a test it should be it's own function/class with accompanying tests.

I would also suggest not using a `regex` for this sort of check as this type of functionality already exists in things like [`QString::toDouble`](http://doc.qt.io/archives/qt-4.8/qstring.html#toDouble) et al or `boost::lexical_cast`. As you already have a `QString` from `toString` I would suggest using the `QString` methods as they will be proven methods of getting edge cases in this type of thing correct.